include/ensmallen_bits/cmaes/active_cmaes_impl.hpp

```diff
@@ -0,0 +1,378 @@
+/**
+ * @file active_cmaes_impl.hpp
+ * @author Marcus Edel
+ * @author Suvarsha Chennareddy
+ *
+ * Implementation of the Active Covariance Matrix Adaptation Evolution Strategy
+ * as proposed by G.A Jastrebski and D.V Arnold in "Improving Evolution
+ * Strategies through Active Covariance Matrix Adaptation".
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_ACTIVE_CMAES_CMAES_IMPL_HPP
+#define ENSMALLEN_ACTIVE_CMAES_CMAES_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "active_cmaes.hpp"
+
+#include <ensmallen_bits/function.hpp>
+
+#ifndef NOT_EMPTY_TRANSFORMATION
+#define NOT_EMPTY_TRANSFORMATION
```

Comment:
not sure what this define is for

Reply:
Thank you for pointing this out! Now that I look at it, I don't think this define is required. The `ACTIVE_CMAES_CMAES_IMPL_HPP` define should be enough. I'll make changes.

Reply:
I'm looking at this once more and I realize that if the define is removed and the user uses both Vanilla and Active CMAES, there will be double inclusion (which is why I added the include guard in the first place). The `ACTIVE_CMAES_CMAES_IMPL_HPP` define is indeed not enough.

Reply:
It sounds like in that case that either `NotEmptyTransfornation` should be moved to its own file and included by both versions, or if the functionality differs here then perhaps it just needs a new name. I think I have understood the problem, let me know if not.

Reply:
Alright, I'll move `NotEmptyTransfornation` to its own file.